examples/core/auth-nextjs-themed/app/(dashboard)/employees/[[...segments]]/page.tsx

```diff
@@ -0,0 +1,84 @@
+'use client';
+import * as React from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { CrudProvider, List, Create, Edit, Show } from '@toolpad/core/Crud';
+import { employeesDataSource, Employee, employeesCache } from '../../../mocks/employees';
+import CustomDataGrid from '../../../components/CustomDataGrid';
+
+function matchPath(pattern: string, pathname: string): string | null {
+  const regex = new RegExp(`^${pattern.replace(/:[^/]+/g, '([^/]+)')}$`);
+  const match = pathname.match(regex);
+  return match ? match[1] : null;
+}
+
+export default function EmployeesCrudPage() {
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const rootPath = '/employees';
+  const listPath = rootPath;
+  const showPath = `${rootPath}/:employeeId`;
+  const createPath = `${rootPath}/new`;
+  const editPath = `${rootPath}/:employeeId/edit`;
+
+  const showEmployeeId = matchPath(showPath, pathname);
+  const editEmployeeId = matchPath(editPath, pathname);
+
+  const handleRowClick = React.useCallback(
+    (employeeId: string | number) => {
+      console.log('Clicked on row with ID', employeeId);
+      router.push(`${rootPath}/${String(employeeId)}`);
+    },
+    [router],
+  );
+
+  const handleCreateClick = React.useCallback(() => {
+    router.push(createPath);
+  }, [createPath, router]);
+
+  const handleEditClick = React.useCallback(
+    (employeeId: string | number) => {
+      router.push(`${rootPath}/${String(employeeId)}/edit`);
+    },
+    [router],
+  );
+
+  const handleCreate = React.useCallback(() => {
+    router.push(listPath);
+  }, [listPath, router]);
+
+  const handleEdit = React.useCallback(() => {
+    router.push(listPath);
+  }, [listPath, router]);
+
+  const handleDelete = React.useCallback(() => {
+    router.push(listPath);
+  }, [listPath, router]);
+
+  return (
+    <CrudProvider<Employee> dataSource={employeesDataSource} dataSourceCache={employeesCache}>
```

Comment:
I'll expose the slot to the top component so we can just use the main `Crud` component here.

Reply:
Feel free to just add a TODO comment for this for now as i guess we will only be able to adjust here after the next release.